### PR TITLE
Add developer guide documentation for builderListComponentCommand

### PR DESCRIPTION
**Pull request by superkaiba**

### CODE REVIEW

File: src/main/java/seedu/bigpp/BigPP.java [hunk 1]
```diff
@@ -22,6 +22,7 @@ public class BigPP {
      * Main entry-point for the java.duke.Duke application.
      */
     public static void main(String[] args) {
+        assert false : "dummy assertion set to fail";
```
Comment (Magmanat) [inline src/main/java/seedu/bigpp/BigPP.java:25]: dummy assertion inside

File: src/main/java/seedu/bigpp/command/buildercommand/BuilderListComponentCommand.java [hunk 2]
```diff
@@ -52,15 +52,15 @@ public String executeCommand(DataStorage dataStorage) throws BuilderMissingListE
...
             throw new BuilderMissingComponentException();
         }
-        if (!dataStorage.stringToComponentListMap.containsKey(componentType)) {
+
+        if (!dataStorage.getStringToComponentListMap().containsKey(componentType)) {
             throw new BuilderIncorrectComponentException();
         }
 
-        ComponentList<?> componentList = dataStorage.stringToComponentListMap.get(componentType);
+        ComponentList<?> componentList = dataStorage.getStringToComponentListMap().get(componentType);
```
Comment (Magmanat) [inline src/main/java/seedu/bigpp/command/buildercommand/BuilderListComponentCommand.java:60]: is this the only place where the change for `getStringToComponentListMap()` is needed?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -7,7 +7,46 @@
...
+The `list [COMPONENT]` command prints out a formatted list of all available components of type `[COMPONENT]`.
+
+When the user inputs a command of the form `list [COMPONENT]` in builder mode, 
+it is parsed by the `Parser.parseBuilderCommand` method which recognizes the first word as
+`list` and creates a new `BuilderListComponentCommand` object while passing as argument the component 
+which was part of the command.
+
+The `BuilderListComponentCommand.executeCommand()` method is then executed as part of the loop
+inside `BigPP.runLoopUntilExit()`. 
+
```
Comment (Magmanat) [inline docs/DeveloperGuide.md:22]: so far looks ok

**Review by Magmanat — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/bigpp/BigPP.java (hunk 1 above)
Comment (superkaiba) [reply src/main/java/seedu/bigpp/BigPP.java]: fixed

File: src/main/java/seedu/bigpp/command/buildercommand/BuilderListComponentCommand.java (hunk 2 above)
Comment (superkaiba) [reply src/main/java/seedu/bigpp/command/buildercommand/BuilderListComponentCommand.java:60]: yes i think so

**Pull request closed without merging**